cofmpy/helpers/extract_fmu.py

```diff
@@ -85,28 +84,162 @@ def extract_fmu_info(fmu_path):
+    """
+
+    console = Console()
+
+    # Check provided output file
+    if output_file is None:
+        console.print("[red]❌ Error: Please provide a valid export file path.[/red]")
+        return
+
+    # Ensure FMU exists
+    if not os.path.isfile(fmu_path):
+        console.print(f"[red]❌ Error: FMU file '{fmu_path}' not found.[/red]")
+        return
+
+    # Extract FMU content
+    unpacked_fmu_dir = fmpy.extract(fmu_path)
+    model_desc = fmpy.read_model_description(fmu_path)
+
+    print(f"Create file {output_file}")
+    with open(output_file, "w", newline="", encoding="utf-8") as csvfile:
+        spam_writer = csv.writer(
+            csvfile, delimiter=",", quotechar="|", quoting=csv.QUOTE_MINIMAL
+        )
+        spam_writer.writerow(
+            ["Category", "name", "type", "type", "unit", "start_value"]
+        )
+
+        # FMU version
+        fmu_version = model_desc.fmiVersion if model_desc.fmiVersion else "N/A"
+        console.print(f"📦 [bold]FMU Version:[/bold] {fmu_version}", style="blue")
+
+        # Integration step size (if available)
+        step_size = (
+            model_desc.defaultExperiment.stepSize
+            if model_desc.defaultExperiment
+            else "N/A"
+        )
+        console.print(
+            f"🕒 [bold]Default Integration Step Size:[/bold] {step_size}\n",
+            style="blue",
+        )
+
+        # Iterate through variables
+        for variable in model_desc.modelVariables:
+            category = (
+                "Parameter"
+                if variable.causality == "parameter"
+                else variable.causality.capitalize()
+            )
+
+            spam_writer.writerow(
+                [
+                    category,
+                    variable.name,
+                    variable.variability,
+                    variable.type,
+                    variable.unit,
+                    str(variable.start if variable.start is not None else "-"),
+                ]
+            )
+
+    # Cleanup extracted FMU folder
+    shutil.rmtree(unpacked_fmu_dir, ignore_errors=True)
+
+    print(f"Information are exported to file {output_file}")
+
+
 def main():
     """
     Main function to parse command-line arguments and call the extraction function.
+
+    command line example to display info from my_fmu.fmu :
+        >> python extract_fmu.py  display my_fmu.fmu"
+    command line example to extract info from my_fmu.fmu and export
+            to extract_infos.csv file :
+        >> python extract_fmu.py export my_fmu.fmu" --output_file "./extract_infos.csv"
     """
     parser = argparse.ArgumentParser(
         description="Extracts and displays FMU information in a structured format."
     )
+    parser.add_argument("action", help="Action to execute, display or export")
     parser.add_argument("fmu_file", help="Path to the FMU file")
+    parser.add_argument("--output_file", help="Path to the output file", required=False)
```

Comment:
To make the CLI arguments simpler, we could only keep the `--output_file` argument that defaults to None. If None, we only display the table (corresponding to `action="display"`). If not None, it saves the table to the CSV file (corresponding to `action="export"`).